smart-home-editor-extension/src/browser/code-generator.ts

```diff
@@ -0,0 +1,65 @@
+import { injectable, inject } from "inversify";
+import URI from "@theia/core/lib/common/uri";
+import { FileSystem } from "@theia/filesystem/lib/common/filesystem";
+import { Workspace } from "@theia/languages/lib/common";
+import { MessageService } from "@theia/core";
+
+@injectable()
+export class CodeGenerator {
+
+    constructor(
+        @inject(FileSystem) protected readonly fileSystem: FileSystem,
+        @inject(Workspace) protected readonly workspace: Workspace,
+        @inject(MessageService) protected messageService: MessageService
+    ) {
+    }
+
+    async readFileContent(uri: URI): Promise<string> {
+        const exists = await this.fileSystem.exists(uri.toString());
+        if (exists) {
+            return this.fileSystem.resolveContent(uri.toString()).then(({ stat, content }) => content);
+        }
+        console.warn('File ' + uri.toString + ' does not exist.');
+        return '';
+    }
+
+    public async generateCode(contentAsString: string, className: string): Promise<void> {
+        if (contentAsString.length > 0) {
+            postRequest(
+                window.location.protocol + '//' + window.location.hostname + ':9091/services/java/fromXMI',
+                contentAsString, 'application/xml')
+                .then(response => {
+                    response.text().then(java => {
+                        const javaURI = new URI(this.workspace.rootUri + "/src/" + className.replace(/\./g, "/") + ".java")
+                        this.setFileContent(javaURI, java, true)
```

Comment:
Maybe it makes sense to not override files as we do not support a merge, yet. If the user wants a freshly generated version, (s)he can simply delete the current files and save the editor.
This would avoid unwanted overriding of code written by the user.

Reply:
Actually, I thought about it again, just leave it like it is and we ask Jonas what he prefers when he's back from vacation. If needed, I can adapt it then.

Reply:
One thing to keep in mind is that we already produce a .java file with our scaffolding project. 
So I think the nice solution would either give the rest service direct write access to the file, so it can do the merging, or sending the contents of the java file also.

Reply:
That's true. Maybe you can open a separate ticket for this after this is merged?